sync-core/src/main/java/com/cloudant/sync/datastore/BasicDatastore.java

```diff
@@ -1250,7 +1301,12 @@ private long insertDocumentHistoryIntoExistingTree(SQLDatabase db, BasicDocument
         options.data = newRevision.asBytes();
         options.available = true;
         options.copyAttachments = false;
-        long sequence = insertRevision(db,options);
+        long sequence;
+        try {
+            sequence = insertRevision(db, options);
+        } catch (AttachmentException e){
+            throw new RuntimeException("Attachments should not be copied when inserting document history",e);
```

Comment:
@tomblench @mikerhodes How does `AttachmentException should not been thrown, this is a library programming error` sound?